wsdl generation output dataclass missing namespace

Here's the wsdl parts for loginResponse

```xml
    <wsdl:message name="loginResponse">
        <part name="Result" type="nas:Result"/>
    </wsdl:message>
    <wsdl:portType name="NetworkManagementApi">
        <wsdl:operation name="login">
            <wsdl:documentation></wsdl:documentation>
            <wsdl:input message="nas:loginRequest"/>
            <wsdl:output message="nas:loginResponse"/>
            <wsdl:fault name="error" message="nas:nasFaultMessage"/>
        </wsdl:operation>
    </wsdl:binding>
```

Results in 
```python
@dataclass
class LoginResponse:
    class Meta:
        name = "loginResponse"

    result: Optional[Result] = field(
        default=None,
        metadata={
            "name": "Result",
            "type": "Element",
            "namespace": "",
        }
    )
```

but should have `"namespace": "http://microfocus.com/nas/2020/08"` in order for parser to be able to parse response [envelope](https://github.com/brunnels/xsdata-mfna/blob/main/test_login.py#L31).

Maybe I did not state the issue correctly?  

On the main branch I generated the client in the [mfna directory](https://github.com/brunnels/xsdata-mfna/tree/main/mfna) with `xsdata -mll 120 --relative-imports -ss clusters -p mfna mfna-partial.wsdl`

If I run [these tests](https://github.com/brunnels/xsdata-mfna/blob/main/test_login.py) on the main branch, they fail.

If I run the same tests from the [generate-fixes branch](https://github.com/brunnels/xsdata-mfna/tree/generate-fixes) they pass.

[Here are the changes](https://github.com/brunnels/xsdata-mfna/pull/2/files) I made to get them to pass.

With the namespace as an empty string in the `LoginResponse` dataclass metadata the `test_success` test fails with `AssertionError: None is not an instance of <class 'mfna.result.Result'>` because it is unable to parse the loginResponse.

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -2,7 +2,7 @@ exclude: tests/fixtures
 
 repos:
   - repo: https://github.com/asottile/pyupgrade
-    rev: v2.29.0
+    rev: v2.29.1
     hooks:
       - id: pyupgrade
         args: [--py37-plus]
@@ -11,7 +11,7 @@ repos:
     hooks:
       - id: reorder-python-imports
   - repo: https://github.com/ambv/black
-    rev: 21.10b0
+    rev: 21.11b1
     hooks:
       - id: black
   - repo: https://gitlab.com/pycqa/flake8
@@ -42,11 +42,11 @@ repos:
       - id: mypy
         additional_dependencies: [tokenize-rt, types-requests, types-Jinja2, types-click, types-docutils]
   - repo: https://github.com/asottile/setup-cfg-fmt
-    rev: v1.19.0
+    rev: v1.20.0
     hooks:
       - id: setup-cfg-fmt
         args: ["--max-py-version=3.10"]
   - repo: https://github.com/PyCQA/doc8
-    rev: 0.9.1
+    rev: 0.10.1
     hooks:
       - id: doc8

--- a/tests/codegen/handlers/test_attribute_type.py
+++ b/tests/codegen/handlers/test_attribute_type.py
@@ -414,3 +414,21 @@ class AttributeTypeHandlerTests(FactoryTestCase):
         attr = AttrFactory.native(DataType.HEX_BINARY)
         self.processor.update_restrictions(attr, attr.types[0].datatype)
         self.assertEqual("base16", attr.restrictions.format)
+
+    def test_detect_lazy_namespace(self):
+        source = ClassFactory.create(namespace="foo", tag=Tag.ELEMENT)
+        target = ClassFactory.create()
+        attr = AttrFactory.create(namespace="##lazy")
+
+        self.processor.detect_lazy_namespace(source, target, attr)
+        self.assertEqual("foo", attr.namespace)
+
+        attr.namespace = "##lazy"
+        source.namespace = None
+        self.processor.detect_lazy_namespace(source, target, attr)
+        self.assertIsNone(attr.namespace)
+
+        attr.namespace = "##lazy"
+        target.namespace = "foo"
+        self.processor.detect_lazy_namespace(source, target, attr)
+        self.assertEqual("", attr.namespace)

--- a/tests/codegen/mappers/test_definitions.py
+++ b/tests/codegen/mappers/test_definitions.py
@@ -694,7 +694,7 @@ class DefinitionsMapperTests(FactoryTestCase):
                 "arg0", str(DataType.STRING), namespace="", native=True
             ),
             DefinitionsMapper.build_attr(
-                "arg1", build_qname("boo", "cafe"), namespace="", native=False
+                "arg1", build_qname("boo", "cafe"), namespace="##lazy", native=False
             ),
         ]
         self.assertIsInstance(result, Generator)
@@ -714,10 +714,11 @@ class DefinitionsMapperTests(FactoryTestCase):
         mock_create_message_attributes.return_value = attrs
         actual = DefinitionsMapper.build_message_class(definitions, port_type_message)
         expected = Class(
-            qname=build_qname("xsdata", "bar"),
-            status=Status.FLATTENED,
+            qname=build_qname("bar", "bar"),
+            status=Status.RAW,
             tag=Tag.ELEMENT,
             location="foo.wsdl",
+            namespace="bar",
             ns_map=message.ns_map,
             attrs=attrs,
         )

--- a/tests/fixtures/hello/hello.py
+++ b/tests/fixtures/hello/hello.py
@@ -36,6 +36,7 @@ class HelloError:
 class GetHelloAsString:
     class Meta:
         name = "getHelloAsString"
+        namespace = "http://hello/"
 
     arg0: Optional[str] = field(
         default=None,
@@ -50,6 +51,7 @@ class GetHelloAsString:
 class GetHelloAsStringResponse:
     class Meta:
         name = "getHelloAsStringResponse"
+        namespace = "http://hello/"
 
     return_value: Optional[str] = field(
         default=None,

--- a/xsdata/codegen/handlers/attribute_type.py
+++ b/xsdata/codegen/handlers/attribute_type.py
@@ -144,6 +144,7 @@ class AttributeTypeHandler(RelativeHandlerInterface):
             if source.nillable:
                 attr.restrictions.nillable = True
             self.set_circular_flag(source, target, attr_type)
+            self.detect_lazy_namespace(source, target, attr)
 
     @classmethod
     def copy_attribute_properties(
@@ -227,3 +228,26 @@ class AttributeTypeHandler(RelativeHandlerInterface):
 
         if datatype in (DataType.NMTOKENS, DataType.IDREFS):
             attr.restrictions.tokens = True
+
+    @classmethod
+    def detect_lazy_namespace(cls, source: Class, target: Class, attr: Attr):
+        """
+        Override attr namespace with the source namespace when during the
+        initial mapping the namespace detection wasn't possible.
+
+        Case 1: WSDL message part type can be an element, complex or
+        simple type, we can't do the detection during the initial
+        mapping to class objects.
+        """
+        if attr.namespace == "##lazy":
+            logger.warning(
+                "Overriding field type namespace %s:%s (%s)",
+                target.name,
+                attr.name,
+                source.namespace,
+            )
+
+            if not source.namespace:
+                attr.namespace = "" if target.namespace else None
+            else:
+                attr.namespace = source.namespace

--- a/xsdata/codegen/mappers/definitions.py
+++ b/xsdata/codegen/mappers/definitions.py
@@ -248,15 +248,18 @@ class DefinitionsMapper:
     ) -> Class:
         """Step 6.2: Build the input/output message class of an rpc style
         operation."""
-        message_name = text.suffix(port_type_message.message)
-        definition_message = definitions.find_message(message_name)
+        prefix, name = text.split(port_type_message.message)
+
+        definition_message = definitions.find_message(name)
         ns_map = definition_message.ns_map.copy()
+        source_namespace = ns_map.get(prefix)
 
         assert definitions.location is not None
 
         return Class(
-            qname=namespaces.build_qname(definitions.target_namespace, message_name),
-            status=Status.FLATTENED,
+            qname=namespaces.build_qname(source_namespace, name),
+            namespace=source_namespace,
+            status=Status.RAW,
             tag=Tag.ELEMENT,
             location=definitions.location,
             ns_map=ns_map,
@@ -357,8 +360,10 @@ class DefinitionsMapper:
             namespace = part.ns_map.get(prefix)
             type_qname = namespaces.build_qname(namespace, type_name)
             native = namespace == Namespace.XS.uri
-            namespace = "" if part.type else namespace
-
+            # If part has a type it could reference an element or a complex type or
+            # a simple type, we can't make that detection yet, postpone it till the
+            # classes processing.
+            namespace = "##lazy" if part.type else namespace
             yield cls.build_attr(name, type_qname, namespace=namespace, native=native)
 
     @classmethod
@@ -392,6 +397,9 @@ class DefinitionsMapper:
     ) -> Attr:
         """Builder method for attributes."""
         occurs = 1 if default is not None else None
+        if native:
+            namespace = ""
+
         return Attr(
             tag=Tag.ELEMENT,
             name=name,

